### PR TITLE
refactor: use Uint8Array instead of Buffer for binary data representation

### PR DESCRIPTION
**Pull request by kyrylodolynskyi**
I tried to remove the usage of `Buffer` as much as possible. Unfortunately, it is not fully possible because some of our dependencies are based on the `Buffer` api, due to this I didn't bother to create our own replacement for `Buffer.from(..., 'base64')`.

### CODE REVIEW

**Review by amitzur — COMMENTED** [review]
This is a sensitive area in the code, I want to make sure the changes here are (a) correct, (b) don't create a performance regression or bigger memory footprint, and (c) that the implementation which is now browser-compatible will stay this way. Will it be tested through eyes-browser-extension once we move to make it on top of `@applitools/eyes`?

Can we run tests in all packages on this before merging to master.
The code itself seems understandable - wherever there was a Node.js buffer, it is now a Uint8Array (or Uint8ClampedArray, which I assume is needed because of 3rd parties).

What about the case of `base64`? We assume that we will get this format of data only from drivers outside the browser, and this is why it shouldn't be an issue? I'm fine with it, I'd just like to know.

**Comment by kyrylodolynskyi** [comment]
@amitzur I beliave the implementation is correct because it is tested on the image package level. I also want to stress the fact that this change didn't make the package 100% node independent. If you need to run this in the browser you will still have to bundle it with `Buffer` polyfill, as long as we use `png-async` and maybe other deps, we will depend on `Buffer`. So this means that the fact that I didn't change the base64 to buffer transformation is okay.
Regarding performance, I also don't think that this will be a performance issue, because I was extra careful about those changes, and made it in a way that will not cause any copying of the array buffers, so basically, just migrated from buffer-specific api to generic api of js.
Regarding preserving these changes, I will update the tests to use JS' `ArrayBuffer` and `Uint8Array` instead of Node's `Buffer` this way we will be able to guarantee that we work with the proper types and not rely on the `Buffer` instance api.

File: js/packages/image/src/formats/png.ts [hunk 1]
```diff
@@ -1,44 +1,30 @@
...
-    wrapper
-      .pack()
-      .pipe(writable)
-      .on('finish', () => resolve(buffer))
-      .on('error', (err: Error) => reject(err))
-  })
+export async function toPng(image: ImageRaw): Promise<ImageBuffer> {
+  const wrapper = new png.Image({width: image.width, height: image.height})
+  wrapper.data = Buffer.from(image.data)
+  return utils.streams.toBuffer(wrapper.pack())
```
[the remark this answers is not in the document]
Comment (kyrylodolynskyi) [reply js/packages/image/src/formats/png.ts:29]: This is even better. It was just developed in a different context and I never got to apply it here. Just saw a great opportunity.

**Comment by kyrylodolynskyi** [comment]
@amitzur Here is the CI run with all of the relevant packages - https://github.com/applitools/eyes.sdk.javascript1/actions/runs/6246370198.

I also made some changes in other packages to rely less on `Buffer` type. I think this should be a general guideline for our team to not rely on the `Buffer` unless you 100% need it. `Buffer` was created at a time when there was no way to work with binary data, but JS in general already has it (and `Buffer` is become just a subclass of `Uint8Array`) we should write a better JS and simply pursue builtin ways to do things.
The thing about `Buffer` that still makes sense is that it still provides some utility functions that we need and there is no other easy way to archive the same (e.g. working with different encodings, such as base64). So we still going to use it, I just want us to understand when we need it and when we can live with `Uint8Array`.

**Pull request merged**